File: hw/gualotuna/u3/HW29CompositePattern/src/main/java/ec/edu/espe/event/hw29compositepattern/President.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.event.hw29compositepattern;

/**
 *
 * @author  Brayan Gualotuña  
 */
public class President extends Supervisior{
    private static President president = new President();
    
    private President(String aName){
        this();
        name = aName;
    }
    
    private President(){
        super();
        title = "President";
    }
    
    public void stateName(){
        super.stateName();
    }
    
    public static President getPresident(String aName){
        president.name = aName;
        return President.president;
    }
}

File: hw/Vaca/u3/HW29TemplateAndCompsite/HW29CompositePatternJavaScript/President.js
class President extends Supervisior {
    static president = new President();

    constructor(aName) {
        super();
        this.name = aName || "Not assigned yet";
        this.title = "President";
    }

    stateName() {
        super.stateName();
    }

    static getPresident(aName) {
        President.president.name = aName;
        return President.president;
    }
}
